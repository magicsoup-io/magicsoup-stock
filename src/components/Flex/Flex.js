import styled from 'styled-components'
import {
  flexWrap,
  flexDirection,
  alignItems,
  justifyContent,
} from 'styled-system'

import Box from '../Box'

const themed = key => props => props.theme[key]

const Flex = styled(Box)`
  display: flex;
  ${flexWrap}
  ${flexDirection}
  ${alignItems}
  ${justifyContent}
  ${themed('Flex')}
`

/** @component */
export default Flex

Flex.propTypes = {
  ...flexWrap.propTypes,
  ...flexDirection.propTypes,
  ...alignItems.propTypes,
  ...justifyContent.propTypes,
}
