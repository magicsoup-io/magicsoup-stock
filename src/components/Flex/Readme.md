Flex:

```jsx
<Flex>
  <Box
    p={3}
    width={1/2}
    color="white"
    bg="primary">
    Flex
  </Box>
  <Box
    p={3}
    width={1/2}
    color="white"
    bg="secondary">
    Box
  </Box>
</Flex>
```
