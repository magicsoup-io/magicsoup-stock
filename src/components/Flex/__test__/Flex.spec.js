import React from 'react'
import renderer from 'react-test-renderer'

import Flex from '../Flex'

it('renders <Flex /> without throwing', () => {
  const tree = renderer.create(<Flex> I am a Flex</Flex>).toJSON()
  expect(tree).toMatchSnapshot()
})
