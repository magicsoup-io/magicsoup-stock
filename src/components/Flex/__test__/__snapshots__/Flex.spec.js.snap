// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Flex /> without throwing 1`] = `
<div
  className="sc-bdVaJa sc-bwzfXH gjfIPe"
>
   I am a Flex
</div>
`;
